ui/src/features/stage/stage-actions.tsx

```diff
@@ -60,8 +63,51 @@ export const StageActions = ({ stage }: { stage: Stage }) => {
     }
   }, [stage, shouldRefetchFreights]);
 
+  const { data: config } = useQuery(getConfig.useQuery());
+  const argoCDAppsLinks = React.useMemo(() => {
+    const shards = Object.values(config?.argocdShards || {});
+
+    return shards.flatMap((shard) => {
+      return stage.spec?.promotionMechanisms?.argocdAppUpdates.map((argoCD) => ({
+        label: argoCD.appName,
+        url: `${shard.url}/applications/${shard.namespace}/${argoCD.appName}`
+      }));
+    });
+  }, [config, stage]);
```

Comment:
When mapping ArgoCD apps to a shard, we can identify which shard based on the following:
```yaml
kind: Stage
metadata:
  annotations:
    kargo.akuity.io/shard: foo
```

In the normal case, the `kargo.akuity.io/shard` key might be missing entirely. Otherwise the value of the key should be one of the map keys in the GetConfig Argo CD URLs API